mock/db_mock.go

```diff
@@ -0,0 +1,420 @@
+// Code generated by MockGen. DO NOT EDIT.
+// Source: /root/go/pkg/mod/github.com/tendermint/tm-db@v0.6.4/types.go
+
+// Package mock is a generated GoMock package.
+package mock
+
+import (
+	reflect "reflect"
+
+	gomock "github.com/golang/mock/gomock"
+	db "github.com/tendermint/tm-db"
+)
+
+// MockDB is a mock of DB interface.
+type MockDB struct {
```

Comment:
whats mockdb / gomock btw? I'm just not familiar

Reply:
It creates a mock implementation of an interface. Here, I created a mock of the database interface - `MockDb`. It allows controlling the behavior of the mock to enter any code branch for test coverage. I mostly used it to simulate DB errors in this PR  which were difficult to simulate with the mem DB that we normally use for testing